### PR TITLE
feat(website): update the iconography page

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

image

### Potential Breaking Changes

None, the old icons still exist and the `Svg` component too

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-icons/bin/index.js [hunk 1]
```diff
@@ -7,24 +7,39 @@ const {rmSync} = require('fs-extra');
...
+        const componentName = getComponentName(file);
         const fileContent = await fs.readFile(file);
         const tsCode = await svgr.transform(
             fileContent.toString('utf8'),
-            {typescript: true, exportType: 'named', namedExport: componentName},
+            {
+                typescript: true,
+                exportType: 'named',
+                namedExport: componentName,
+                svgProps: {className: '{"icon mod-stroke " + props.className}', },
```
Comment (gdostie) [inline packages/react-icons/bin/index.js:38]: This means all icons now have `icon` `mod-stroke` classes by default

**Comment by aiheon** [comment]
The demo freeze instantly on my end, I cannot use the website :sweat_smile: 
And it's not just my internet, mes others pages are fine and fluid...

File: packages/react-icons/bin/index.js (hunk 1 above)
Comment (aiheon) [reply packages/react-icons/bin/index.js:38]: won't this create conflicts with the style if we add mod-fill? :eyes:

File: packages/website/src/pages/foundations/Iconography.tsx [hunk 2]
```diff
@@ -1,34 +1,102 @@
...
-                .map((svgName: SvgName) => (
-                    <IconItem key={svgName} svgName={svgName} fileName={Icons[svgName].fileName} />
-                ))}
-        </ul>
-    </PlasmaComponent>
+const {iconsList, ...Icons} = PlasmaReactIcons;
+
+const Sandbox = dynamic(
+    import('../../building-blocs/Sandbox').then((mod) => mod.Sandbox),
+    {ssr: false, loading: () => <PlasmaLoading />}
```
Comment (aiheon) [inline packages/website/src/pages/foundations/Iconography.tsx:14]: I really like the built-in loading prop in Next :eyes: 
So much cleaner that the if(loading) {return <Loading />} else { return <MyPage />}

File: packages/react-icons/bin/index.js (hunk 1 above)
Comment (gdostie) [reply packages/react-icons/bin/index.js:38]: `mod-fill`? this class doesn't exist 🤔

**Review by GermainBergeron — APPROVED** [review]

File: packages/website/src/pages/foundations/Iconography.tsx [hunk 3]
```diff
@@ -1,34 +1,102 @@
...
+
+export const IconographyExamples = () => (
+    <>
+        <PageHeader
+            section="Foundations"
+            title="Iconography"
+            thumbnail="iconography"
+            description="Icons are used to visually represent actions, functionalities, and features."
+        />
+        <div className="palsma-page-layout">
```
Comment (GermainBergeron) [inline packages/website/src/pages/foundations/Iconography.tsx:91]: I don't know if we have the same typo elsewhere?
```suggestion
        <div className="plasma-page-layout">
```

**Review by GermainBergeron — CHANGES_REQUESTED** [review]
Hmm after looking into the demo I'm not sure we should add `icon mod-stroke` otherwise the `stroke: currentColor` doesn't work. 

image

If I remove the icon class the size is wrong but the colour is correct

image

File: packages/website/src/pages/foundations/Iconography.tsx [hunk 4]
```diff
@@ -1,34 +1,101 @@
...
+    import * as React from 'react';
+    import { InfoSize32Px } from '@coveord/plasma-react-icons';
+
+    
+    // Control the size using "height" or "width" attributes (defaults to 1em)
+    // The icon takes the same color as the text around it
+
+    export default () => (
+        <div style={{color: 'var(--info-60)'}}>
+            <InfoSize32Px height={64} />;
```
Comment (aiheon) [inline packages/website/src/pages/foundations/Iconography.tsx]:
```suggestion
            <InfoSize32Px height={64} />

```
I think it's this one :see_no_evil: 
image

**Review by aiheon — APPROVED** [review]

**Review by GermainBergeron — APPROVED** [review]

**Pull request merged**